### PR TITLE
Hw2

### PR DESCRIPTION
**Pull request by ELondonJ**
Solutions for homework 2

### CODE REVIEW

File: hw2/01-coins.js [hunk 1]
```diff
@@ -2,47 +2,92 @@
...
+          string += 'pennies';
+        } else {
+          string += key;
+        }
+      } else {
+        string += key;
+        if (value > 1) {
+          string += 's';
+        }
+        string += ', ';
```
Comment (Vincentmask) [inline hw2/01-coins.js:57]: Looks great since we only need to handle the penny to pennies case for plural here

File: hw2/04-api.js [hunk 2]
```diff
@@ -1,7 +1,39 @@
...
+    data.sort((a, b) => {
+      const ca = a.name.common;
+      const cb = b.name.common;
+
+      if (ca < cb) return -1;
+      if (ca > cb) return 1;
+
+      return 0;
+    });
+    data.forEach((country) => {
```
Comment (Vincentmask) [inline hw2/04-api.js:29]: clean work for sorting and appending

File: hw2/01-coins.js [hunk 3]
```diff
@@ -1,7 +1,65 @@
...
+  }
+  if (input > 10.0) {
+    string += 'Error: the number is too large';
+    return string;
+  }
+
+  let change = input;
+  const dollars = Math.floor(input);
+  const money = {
+    dollar: dollars,
```
Comment (CristianTS599) [inline hw2/01-coins.js:19]: "dollar: dollars,"  I believe can just be "dollars," unless the plural aspect vs not is important to functionality [See objects section 3.4](https://github.com/airbnb/javascript#objects)

Everything else about the object looks good your properties are defined well and none of them are quoted

File: hw2/01-coins.js [hunk 4]
```diff
@@ -1,7 +1,65 @@
-/** Exercise 01 - Coins **/
+/** Exercise 01 - Coins * */
 
 const calculateChange = (input) => {
-  // Add your code here
+  let string = ` $${input} ===> `;
+  const verify = /^\d+(?:\.\d{0,2})$/;
```
Comment (CristianTS599) [inline hw2/01-coins.js:5]: Good job on the regexp I tested a few cases and it seemingly works good I think this is a great way to check precision.

File: hw2/01-coins.js [hunk 5]
```diff
@@ -1,7 +1,65 @@
-/** Exercise 01 - Coins **/
+/** Exercise 01 - Coins * */
 
 const calculateChange = (input) => {
-  // Add your code here
+  let string = ` $${input} ===> `;
```
Comment (CristianTS599) [inline hw2/01-coins.js:4]: Not a big deal but maybe giving the variable "string" a more descriptive name. [See Naming Conventions Section 23.1](https://github.com/airbnb/javascript#naming-conventions)

File: hw2/02-reverse.js [hunk 6]
```diff
@@ -1,3 +1,32 @@
...
+  }
+  if (number.toString(10).length !== 8) {
+    const inValid = document.createElement('div');
+    inValid.id = 'invalid';
+    inValid.classList = 'mt-4 text-danger';
+    inValid.innerHTML = 'Error: Please input an 8-digit number';
+    main.appendChild(inValid);
+    return;
+  }
+  const reversed = number
```
Comment (CristianTS599) [inline hw2/02-reverse.js:21]: nice one line reverse

File: hw2/03-form.html [hunk 7]
```diff
@@ -1,19 +1,38 @@
...
+    <div class="form-group my-2">
+      <label class="form-group my-2" for="userEmail">Email*</label>
+      <input type="email" class="form-control" id="userEmail" name="email">
+      <div class="form-group my-3">
+        <label class="form-group my-2" for="feedbackMessage">Submit your message</label>
+        <textarea class="form-control" id="feedbackMessage" name="feedback" spellcheck="false"></textarea>
+      </div>
+      <div class="form-group form-check my-2">
+        <input type="checkbox" class="form-check-input" id="exampleCheck1" name="newsletter">
+        <label class="form-check-label" for="exampleCheck1">Sign up for the newsletter</label>
```
Comment (CristianTS599) [inline hw2/03-form.html:29]: Good job on accessibility

File: hw2/03-form.js [hunk 8]
```diff
@@ -1,3 +1,24 @@
...
+  if (fv.get('name') !== '') name = fv.get('name');
+  if (fv.get('email') !== '') email = fv.get('email');
+  if (fv.get('feedback') !== '') feedback = fv.get('feedback');
+  if (fv.get('newsletter') === null) newslet = noNewslet;
+
+  console.log('=======Form Submission=======');
+  console.log(`\tName: ${name}`);
+  console.log(`\tEmail: ${email}`);
+  console.log(`\tFeedback: ${feedback}`);
+  console.log(`\tNewsletter: ${newslet}`);
```
[the remark this answers is not in the document]
Comment (CristianTS599) [reply hw2/03-form.js:23]: Yes, very nice good job.

**Review by CristianTS599 — COMMENTED** [review]
Good work overall not much airbnb guide stuff to recommend and your code has a very clean and organized look.
I checked your variable case / naming, object structuring and use, and accessibility against the guide and it all looks good.

**Comment by JeremyIV** [comment]
Exercise 1 looks good. 

in Exercise 2, name and email inputs should be marked with the 'required' attribute. Otherwise looks good.

For ex4, if the API request fails, it would be better to display an error message on the webpage rather than just in the console. Otherwise, looks good.

**Pull request merged**